flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -598,189 +535,154 @@ protected int cancel(String[] args) {
 	 */
-	protected int savepoint(String[] args) {
+	protected int savepoint(String[] args) throws CliArgsException {
 		LOG.info("Running 'savepoint' command.");
 
-		SavepointOptions options;
-		try {
-			options = CliFrontendParser.parseSavepointCommand(args);
-		} catch (CliArgsException e) {
-			return handleArgException(e);
-		} catch (Throwable t) {
-			return handleError(t);
-		}
+		SavepointOptions options = CliFrontendParser.parseSavepointCommand(args);
 
 		// evaluate help flag
 		if (options.isPrintHelp()) {
 			CliFrontendParser.printHelpForSavepoint();
 			return 0;
 		}
 
-		if (options.isDispose()) {
-			// Discard
-			return disposeSavepoint(options);
-		} else {
-			// Trigger
-			String[] cleanedArgs = options.getArgs();
-			JobID jobId;
+		CustomCommandLine<?> customCommandLine = getActiveCustomCommandLine(options.getCommandLine());
 
-			if (cleanedArgs.length >= 1) {
-				String jobIdString = cleanedArgs[0];
-				try {
-					jobId = new JobID(StringUtils.hexStringToByte(jobIdString));
-				} catch (Exception e) {
-					return handleArgException(new IllegalArgumentException(
-							"Error: The value for the Job ID is not a valid ID."));
-				}
+		ClusterClient clusterClient = customCommandLine.retrieveCluster(options.getCommandLine(), config, configurationDirectory);
+
+		try {
+			if (options.isDispose()) {
+				// Discard
+				return disposeSavepoint(clusterClient, options.getSavepointPath());
 			} else {
-				return handleArgException(new IllegalArgumentException(
-						"Error: The value for the Job ID is not a valid ID. " +
-								"Specify a Job ID to trigger a savepoint."));
-			}
+				// Trigger
+				String[] cleanedArgs = options.getArgs();
+				JobID jobId;
+
+				if (cleanedArgs.length >= 1) {
+					String jobIdString = cleanedArgs[0];
+					try {
+						jobId = new JobID(StringUtils.hexStringToByte(jobIdString));
+					} catch (Exception ignored) {
+						throw new CliArgsException("Error: The value for the Job ID is not a valid ID.");
+					}
+				} else {
+					throw new CliArgsException("Error: The value for the Job ID is not a valid ID. " +
+						"Specify a Job ID to trigger a savepoint.");
+				}
 
-			String savepointDirectory = null;
-			if (cleanedArgs.length >= 2) {
-				savepointDirectory = cleanedArgs[1];
-			}
+				String savepointDirectory = null;
+				if (cleanedArgs.length >= 2) {
+					savepointDirectory = cleanedArgs[1];
+				}
 
-			// Print superfluous arguments
-			if (cleanedArgs.length >= 3) {
-				logAndSysout("Provided more arguments than required. Ignoring not needed arguments.");
-			}
+				// Print superfluous arguments
+				if (cleanedArgs.length >= 3) {
+					logAndSysout("Provided more arguments than required. Ignoring not needed arguments.");
+				}
 
-			return triggerSavepoint(options, jobId, savepointDirectory);
+				return triggerSavepoint(clusterClient, jobId, savepointDirectory);
+			}
+		} catch (Exception e) {
+			return handleError(e);
```

Comment:
Why catch exception here? It's already done in `parseParameters(String[] args)`.

Reply:
Oversight from my side.